src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/MIReplicaReportingServiceTest.java

```diff
@@ -319,6 +319,75 @@ void should_save_task_and_get_transformed_state_label_from_reportable_task(
                 });
     }
 
+    @ParameterizedTest
+    @CsvSource(value = {
+        "ASSIGNED,AutoAssign,true,true",
+        "UNASSIGNED,Configure,false,true",
+        "COMPLETED,AutoAssign,false,false",
+        "CANCELLED,Configure,false,false"
```

Comment:
Are there a few more permutations here?  e.g. ASSIGNED,AutoAssign,false,false?
Could we also have cases where the state is COMPLETED and there is already a report there?

Reply:
Hello Ravi, In this test, I am not maintaining state transitions, but directly creating tasks with final State and final Update Action. Just to test if the task_assignments gates are working.